### PR TITLE
GET /api/articles (queries) for review

### PR DESCRIPTION
**Pull request by Chris-Mark-Wilson**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
...
+      .get("/api/articles?topic=")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+      });
+  });
+  it("should 404 for non existent topic", () => {
+    return request(app)
+      .get("/api/articles?topic=the flying peshwaris")
```
Comment (dappernerddesigns) [inline __tests__/integration.test.js]: I'm sure this is giving the correct 404. Just keep an eye on spaces in endpoints. Not an issue, just be aware.

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
...
+      .expect("Content-Type", /json/)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+  it("should 404 for bad datatype", () => {
+    return request(app)
```
Comment (dappernerddesigns) [inline __tests__/integration.test.js]: Is a 404 an appropriate status code for a bad datatype? If this is a bad topic search this should be a 400.

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
...
+      .get("/api/articles?sort_by=topic")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+        expect(articles).toBeSortedBy("topic", { descending: true });
+      });
+  });
+  it("should sort by a given query", () => {
```
Comment (dappernerddesigns) [inline __tests__/integration.test.js]: Could these test descriptions be a bit more descriptive? And is each sort_by a new behaviour? If it isn't a new behaviour, does it need a new test?

File: __tests__/integration.test.js [hunk 4]
```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
...
+  it("should 400 for an invalid order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at&order=dodgy")
+      .expect("Content-Type", /json/)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("DODGY is not a valid order use [ASC, DESC]");
+      });
+  });
+});
```
Comment (dappernerddesigns) [inline __tests__/integration.test.js:603]: I think there is one test missing here, a 200 for a completely valid sort_by with no articles :)

File: models/articles_models.js [hunk 5]
```diff
@@ -14,34 +14,66 @@ const selectArticleById = (id) => {
...
+  }
+
+  baseQuery += `GROUP BY articles.author,articles.title,articles.article_id
+   ORDER BY ${sort_by} ${order};`;
+
+  return db.query(baseQuery, queryValues).then(({ rows }) => {
+    if (rows.length === 0)
+      return Promise.reject({ status: 404, msg: "not found" });
+    else return rows;
+  });
```
Comment (dappernerddesigns) [inline models/articles_models.js:57]: Consider a valid topic that has no articles, is that a 404?

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
This is very close! I'd like you to look at the comments that I've left you in regards to a missing test and your test descriptions and then re submit please 😄

**Review by mezzDavies — APPROVED** [review]
Nice. Good to merge 👍

**Pull request merged**